feat: aztec-cli "unbox" command (broken - closing)

dan-aztec opened the pull request:
WIP, not working or ready for full review.  

shows my high level thoughts on implementation approach based on https://github.com/AztecProtocol/aztec-packages/issues/1056

feedback thus far:
(1) instead of looking up the install directory from `npm i -g...` for the aztec-cli package, let's download the `noir-contracts` from github directly (done)
(2) setup a new subpackage `starter-kit` for the barebones webserver + frontend (done, but it's basically the default next.js project.  tried to merge its settings with our other subpackage package.json)

open questions on my side:
fix compiling.  does the copied Nargo.toml need to be updated?  it's doing a relative path to `noir-libs`
resolved questions:
does this require a local sandbox running as well? (yes)
why aren't the formik forms working on the frontend?  (css issue: z-index was too far behind)

# Checklist:
Remove the checklist to signal you've completed it. Enable auto-merge if the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g. cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this pull request to relevant issues (if any exist).

sourdzl committed:
initial sketch

sourdzl committed:
forgot new file

sourdzl committed:
comment

sourdzl committed:
sign commit

dan-aztec committed:
try to sign

dan-aztec committed:
Merge branch 'master' into dan/aztec-cli-unbox

dan-aztec committed:
download from github

dan-aztec committed:
add starter-kit subpackage

dan-aztec committed:
remove comments

dan-aztec committed:
Merge branch 'master' into dan/aztec-cli-unbox

dan-aztec committed:
add aztec banner

fix colorscheme

fix banner

aztec logo

prettier working?

add api endpoint

switch to cjs file

parse contract functions, but no working form

delete unneeded text

comments

slightly cleaner form

slightly better css

fix: Remove automatic update to `AztecProtocol/dev-bb.js` (#1712)

Related to #1711

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

git subrepo push --branch=master circuits/cpp/barretenberg

subrepo:
  subdir:   "circuits/cpp/barretenberg"
  merged:   "d883900f9"
upstream:
  origin:   "https://github.com/AztecProtocol/barretenberg"
  branch:   "master"
  commit:   "d883900f9"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

chore: sync bb master (#1713)

Ran:

```
./scripts/git_subrepo.sh pull circuits/cpp/barretenberg
git checkout origin/master -- .gitmodules
```

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

copy the contract ABI into the new subdirectory

fix output json name

dan-aztec committed:
Merge branch 'master' into dan/aztec-cli-unbox

dan-aztec committed:
comments

fix docstring

update new README

add note on running sandbox

fix css on form to make usable

add new folders

better contract not found message

let command copy to optional target directory

also create a .env file in new folder

fix starter kit directory copying

add aztec-cli dep

migrate to vite from create-next-app

remove .next

revert top level yarn.lock change

fix width overflow

dan-aztec commented on yarn-project/starter-kit/README.md:
```diff
@@ -0,0 +1,34 @@
+This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
```
will change this README

dan-aztec commented:
i have been testing the `starter-kit` subpackage by coping the `private_token_contract.json` ABI in `noir-contracts/artifacts` into `starter-kit/noir-contracts`, not 100% sure about the package management here

dan-aztec commented:
TODO: look at https://github.com/noir-lang/noir-starter/tree/main/next-hardhat

LeilaWang commented:
Some thoughts:

When users run the command:
`aztec-cli unbox private_token [optional_name_of_the_output_dir]`

They will find the following files:
```
private_token
|— README.md
|— package.json
|— any.config.cjs
|— src
       |— app
              |— index.tsx
              |— globals.css
              |— Contract.tsx
              |— DeployForm.tsx
              |— FnForm.tsx
       |- scripts
              |— index.tx
              |— deploy_contract.tx
              |— call_contract_function.tx
       |— contracts
              |— private_token_contract
                     |— main.nr
                     |— Cargo.toml
       |— artifacts
              |— private_token_contract.json
              |— private_token.ts
       |— test
```

#### `package.json`

It should provide scripts to compile noir contracts, run tests, and start the website.

```json
"scripts": {
    "compile": "aztec-compile compile --outdir ./src/artifacts --typescript ./src/artifacts",
    "test": "jest",
    "start": "next start"
}
```

### App

The app speaks to the AztecRPCServer (which should already be running). We don't need a web server for the frontend.

#### `Contract.tsx`

We can import the already generated abi and render a form for each contract function:

```tsx
import PrivateTokenJson from '../artifacts/private_token_contract.json' assert { type: 'json' };

<DeployForm abi={constructorAbi} onDeployed={contractAddress => setState('contractAddress', contractAddress)} />
{!!state.contractAddress && (
  PrivateTokenJson.functions.filter().map(fnAbi => <FnForm contractAddress={state.contractAddress} abi={fnAbi}></FnForm>)
)}
```

#### `DeployForm.tsx`

We should create a file in `scripts` to demonstrate how to do something.

```tsx
import { deployContract } from '../scripts/deploy_contract.js';

onSubmit: deployContract(contractAbi, inputs)
```

#### `FnForm.tsx`

```tsx
import { callContractFunction } from '../scripts/call_contract_function.js';

onSubmit: (inputs) => callContractFunction(contractAddress, fnAbi, inputs)
```

#### `scripts/call_contract_function.ts`

```js
function callContractFunction(contractAddress, fnAbi, inputs) {
       if (fnAbi.functionType === 'unconstrained') {
              viewTx();
       } else {
              sendTx();
       }
}
```

### Nice to have

####  `Account.tsx`

Fetch accounts from RPCServer and display them here. Users can pick which one to use, or create a new account.

All open to discussions :)

dan-aztec committed:
Merge branch 'master' into dan/aztec-cli-unbox

dan-aztec committed:
expose env vars to vite

dan-aztec committed:
copy noir-libs and update copied nargo.toml

refactor: consistent block number method naming (#1751)

Renamed `AztecRPC.getBlockNum` to `getBlockNumber`
and`AztecNode.getBlockHeight` to `getBlockNumber`. I decided to use the
name block number because in the Ethereum JSON RPC spec there is
`eth_blockNumber` call and I think it's a good idea to use the same
naming.

git subrepo push --branch=main docs

subrepo:
  subdir:   "docs"
  merged:   "324402a78"
upstream:
  origin:   "https://github.com/AztecProtocol/docs"
  branch:   "main"
  commit:   "324402a78"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

refactor: Use context instead of custom oracles for public functions (#1754)

Fixes #1753, #1755 and use context for nullifiers and commitments in
public.

docs: convert quick start guides into e2e tests (#1726)

Fixes #1564

git subrepo push --branch=main docs

subrepo:
  subdir:   "docs"
  merged:   "ba5d7a6bc"
upstream:
  origin:   "https://github.com/AztecProtocol/docs"
  branch:   "main"
  commit:   "ba5d7a6bc"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

feat(ci): Initial release please config (#1769)

PR with initial release please configuration

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

docs: including "real" code in keys docs (#1767)

Including "real" code in keys docs + addressed one Noir TODO so that it
doesn't get shown inside the docs.

git subrepo push --branch=main docs

subrepo:
  subdir:   "docs"
  merged:   "842a54250"
upstream:
  origin:   "https://github.com/AztecProtocol/docs"
  branch:   "main"
  commit:   "842a54250"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

feat: not retrying unrecoverable errors (#1752)

Fixes #1511
Fixes #1724

With this PR all the errors thrown in the server code are considered to
be unrecoverable. Recoverable errors should not be errors and should be
handled (or shown only as warnings). For example I refactored the
`registerAccount` and `registerRecipient` to not throw if we add the
same recipient/account twice because that situation is easily
recoverable (just ignore it).

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

feat: compress debug symbols (#1760)

Partial work towards #1224

While working on brillig debug symbols I noticed that the JSON ABIs
started to weight more than some megabytes and started to create issues
for the typescript type inference, since we're importing them as json
modules. This PR addresses that by just compressing the debug symbols
and decompressing them transparently in the utility function that we
have for this in foundation. I used https://www.npmjs.com/package/pako
for gzip since it should be compatible with the browser without issue.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [x] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [x] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

ci: Updated release please config (#1773)

This PR contains further release-please configuration changes.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

feat(bb): Use an environment variable to set the transcript URL (#1750)

Related to #1749

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

git subrepo push --branch=master circuits/cpp/barretenberg

subrepo:
  subdir:   "circuits/cpp/barretenberg"
  merged:   "41d362e9c"
upstream:
  origin:   "https://github.com/AztecProtocol/barretenberg"
  branch:   "master"
  commit:   "41d362e9c"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

chore(ci): Updated release please config (#1775)

This PR provides further release-please configuration changes.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

chore(ci): set up nightly barretenberg releases (#1761)

This PR pulls across the publishing workflow from Noir so that
barretenberg can have nightly releases.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [x] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [x] Every change is related to the PR description.
- [x] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

chore(master): release 0.1.0-alpha45 (#1774)

:robot: I have created a new Aztec Packages release
---

[0.1.0-alpha45](https://github.com/AztecProtocol/aztec-packages/compare/v0.1.0-alpha44...v0.1.0-alpha45)
(2023-08-23)

* **bb:** Use an environment variable to set the transcript URL
([#1750](https://github.com/AztecProtocol/aztec-packages/issues/1750))
([31488c1](https://github.com/AztecProtocol/aztec-packages/commit/31488c19acfdfd5ff0c3e7f242f94dc0aa049158))
* **ci:** Initial release please config
([#1769](https://github.com/AztecProtocol/aztec-packages/issues/1769))
([4207559](https://github.com/AztecProtocol/aztec-packages/commit/42075590058b21f38b5e745af54b2062371f9ebe))
* compress debug symbols
([#1760](https://github.com/AztecProtocol/aztec-packages/issues/1760))
([9464b25](https://github.com/AztecProtocol/aztec-packages/commit/9464b25c1a2a809db559ddc4e2d4ee5ade1fa65a))
* not retrying unrecoverable errors
([#1752](https://github.com/AztecProtocol/aztec-packages/issues/1752))
([c0f2820](https://github.com/AztecProtocol/aztec-packages/commit/c0f28204f53152c941704ece66287eddfe13c047))

* Download SRS using one canonical URL across the codebase
([#1748](https://github.com/AztecProtocol/aztec-packages/issues/1748))
([899b055](https://github.com/AztecProtocol/aztec-packages/commit/899b05557365a5bf97e64793dd563a1b4bfa0f3f))
* proving fails when circuit has size &gt; ~500K
([#1739](https://github.com/AztecProtocol/aztec-packages/issues/1739))
([708b05c](https://github.com/AztecProtocol/aztec-packages/commit/708b05ca6638dc0d6ca7cb34fb8de76665a43b58))

* **ci:** set up nightly barretenberg releases
([#1761](https://github.com/AztecProtocol/aztec-packages/issues/1761))
([e0078da](https://github.com/AztecProtocol/aztec-packages/commit/e0078dabfcd9e006c2a489c7142ab141d5d81b80))
* **ci:** Updated release please config
([#1775](https://github.com/AztecProtocol/aztec-packages/issues/1775))
([0085e8b](https://github.com/AztecProtocol/aztec-packages/commit/0085e8b17efc36256974f82525530c39ed182639))
* consistent block number method naming
([#1751](https://github.com/AztecProtocol/aztec-packages/issues/1751))
([df1afe2](https://github.com/AztecProtocol/aztec-packages/commit/df1afe255d3095a9b2851b47480801c06d116eed))
* Use context instead of custom oracles for public functions
([#1754](https://github.com/AztecProtocol/aztec-packages/issues/1754))
([46de77a](https://github.com/AztecProtocol/aztec-packages/commit/46de77ad3e5e91b9276146410381c69ccba1ae2b))

* convert quick start guides into e2e tests
([#1726](https://github.com/AztecProtocol/aztec-packages/issues/1726))
([802a678](https://github.com/AztecProtocol/aztec-packages/commit/802a678e3dd19339cd88b105a0ce341026b58054)),
closes
[#1564](https://github.com/AztecProtocol/aztec-packages/issues/1564)
* including "real" code in keys docs
([#1767](https://github.com/AztecProtocol/aztec-packages/issues/1767))
([cd9cadb](https://github.com/AztecProtocol/aztec-packages/commit/cd9cadbfb6b0311c381586799588a5f64df98f29))

---
This PR was generated with [Release
Please](https://github.com/googleapis/release-please). See
[documentation](https://github.com/googleapis/release-please#release-please).

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

chore: sync bb master (#1776)

Ran:

```
./scripts/git_subrepo.sh pull circuits/cpp/barretenberg
git checkout origin/master -- .gitmodules
```
Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

git_subrepo.sh: Fix parent in .gitrepo file.

git subrepo push --branch=master circuits/cpp/barretenberg

subrepo:
  subdir:   "circuits/cpp/barretenberg"
  merged:   "1b1d24e82"
upstream:
  origin:   "https://github.com/AztecProtocol/barretenberg"
  branch:   "master"
  commit:   "1b1d24e82"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

update noir contract paths

feat: CDP/Lending example contract (#1554)

Finishing up the lending/cdp contract enough for show (no liqudation and unsecure as all the contracts). See #1460.

more instructions

also reference the generated typescript file

start frontend integration w/sandbox

cleaner parsing of functionAbi for yup schema

re-add initialValues

switch to rpcclient instead of server

hardcode some private keys from fixtures

try singleKeyAccount

thanks adam for fixing webasm import

switch to vite.config.js

switch to privateKey class

blocked by undefined methods attribute on the PrivateTokenContract object

revert yarn.lock osx change

use latest yarn.lock

docs: events (#1768)

Fixes #1756

git subrepo push --branch=main docs

subrepo:
  subdir:   "docs"
  merged:   "3c5f3c4a9"
upstream:
  origin:   "https://github.com/AztecProtocol/docs"
  branch:   "main"
  commit:   "3c5f3c4a9"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

feat: no unencrypted logs in private functions (#1780)

Fixes #1689

git subrepo push --branch=main docs

subrepo:
  subdir:   "docs"
  merged:   "40c05467f"
upstream:
  origin:   "https://github.com/AztecProtocol/docs"
  branch:   "main"
  commit:   "40c05467f"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

chore(ci): Updated release please configuration (#1787)

This PR contains further configuration changes and documentation for our
usage of release please

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

chore(master): release 0.1.0-alpha46 (#1777)

:robot: I have created a new Aztec Packages release
---

[0.1.0-alpha46](https://github.com/AztecProtocol/aztec-packages/compare/v0.1.0-alpha45...v0.1.0-alpha46)
(2023-08-24)

* CDP/Lending example contract
([#1554](https://github.com/AztecProtocol/aztec-packages/issues/1554))
([ecf6df2](https://github.com/AztecProtocol/aztec-packages/commit/ecf6df201047dcaa61c270cdb512cdc62086b356))
* no unencrypted logs in private functions
([#1780](https://github.com/AztecProtocol/aztec-packages/issues/1780))
([4d8002e](https://github.com/AztecProtocol/aztec-packages/commit/4d8002e0d101a14c465929d92ea05d0be6e8d99a)),
closes
[#1689](https://github.com/AztecProtocol/aztec-packages/issues/1689)

* **ci:** Updated release please configuration
([#1787](https://github.com/AztecProtocol/aztec-packages/issues/1787))
([6eb2f7a](https://github.com/AztecProtocol/aztec-packages/commit/6eb2f7abc40bae88ebeec546ad9f8f2c7d810a24))
* sync bb master
([#1776](https://github.com/AztecProtocol/aztec-packages/issues/1776))
([7c6fb15](https://github.com/AztecProtocol/aztec-packages/commit/7c6fb15979b48d4d4d5eb5a1ea83d3c0d0ee3b5e))

* events
([#1768](https://github.com/AztecProtocol/aztec-packages/issues/1768))
([5a38cea](https://github.com/AztecProtocol/aztec-packages/commit/5a38cea3f7c1567a8eea3d6c2c58cad6f79b05f2)),
closes
[#1756](https://github.com/AztecProtocol/aztec-packages/issues/1756)

---
This PR was generated with [Release
Please](https://github.com/googleapis/release-please). See
[documentation](https://github.com/googleapis/release-please#release-please).

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

chore: Add todo for using generator indices in note commitment and nullifier computation. (#1762)

fix(noir): Add workaround for latest noir in account contracts (#1781)

Workaround for this issue https://github.com/noir-lang/noir/issues/2421
so we can update the aztec tag to master.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [x] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

chore: split out yarn-project bootstrap.sh (#1790)

Allows for more modular bootstrapping.

chore(p2p): Updated libp2p dependencies (#1792)

This PR simply updates the libp2p dependencies to the newest versions.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

feat: `FunctionSelector` type (#1518)

Fixes #1424

chore: Sandbox logging tweaks (#1797)

Packages a bunch of tweaks to Sandbox debugging and logging, such as:
- Wasm debug logs are now prefixed as `aztec:wasm`, not `wasm`, so they
are visible when debugging via `aztec:*`
- Defaults sandbox logging to INFO instead of DEBUG
- Allows users to configure sandbox debug by exporting `DEBUG='aztec:*'`
in their shell (related to #1605)
- Silences all anvil logs since they didn't provide any useful info
(fixes #1580)
- Renames container names in the sandbox docker-compose (anvil was not
running a fork, and the sandbox is not just an rpc-server)

fix: increment time by 1 for previous rollup was warped (#1594)

With Warp
```
L2 block 1: occurred at t = 100.
Call warp(200) => Rollup.sol's lastBlockTs = 200 & L1.setNextBlockTimeStamp = 200.
L2 block 2: txs show t = 200. Rollup published at t = 200 => Rollup.sol's lastBlockTs = 200
L2 block 3: txs show t = 200.
```
Notice how txs in block 2 and block 3 show a timestamp of 200! This is
confusing.

So we check if the last rollup was warped (here block 2), and if so, txs
in the next rollup (block 3) should show ts = 201. We check if last
rollup was warped by introducing a variable in Rollup.sol that tracks
the last time block was warped.

Also Create #1614

fix: selector name regression (#1800)

I introduced a regression in my function [selector type
PR](https://github.com/AztecProtocol/aztec-packages/pull/1518) which
caused the selector name to be incorrect in circuits.gen.ts. The issue
was with having different names for selector in FunctionData struct in
TS and C++.

This PR fixes it.

chore(master): release 0.1.0-alpha47 (#1788)

:robot: I have created a new Aztec Packages release
---

[0.1.0-alpha47](https://github.com/AztecProtocol/aztec-packages/compare/v0.1.0-alpha46...v0.1.0-alpha47)
(2023-08-25)

* `FunctionSelector` type
([#1518](https://github.com/AztecProtocol/aztec-packages/issues/1518))
([942f705](https://github.com/AztecProtocol/aztec-packages/commit/942f7058adc706924ff26d2490bec7f7d57d7149)),
closes
[#1424](https://github.com/AztecProtocol/aztec-packages/issues/1424)

* increment time by 1 for previous rollup was warped
([#1594](https://github.com/AztecProtocol/aztec-packages/issues/1594))
([2a52107](https://github.com/AztecProtocol/aztec-packages/commit/2a521070397b6d1915e55b4ec702d4778563e683))
* **noir:** Add workaround for latest noir in account contracts
([#1781](https://github.com/AztecProtocol/aztec-packages/issues/1781))
([eb8a052](https://github.com/AztecProtocol/aztec-packages/commit/eb8a052ad4e19394f096cc3a0f533c2560a7f5cc))
* selector name regression
([#1800](https://github.com/AztecProtocol/aztec-packages/issues/1800))
([a5be8bb](https://github.com/AztecProtocol/aztec-packages/commit/a5be8bb92f858d266cf96671c46343b6e1ff400a))

* Add todo for using generator indices in note commitment and nullifier
computation.
([#1762](https://github.com/AztecProtocol/aztec-packages/issues/1762))
([2db6728](https://github.com/AztecProtocol/aztec-packages/commit/2db6728fcaf75ce8c98d821b65695543bb0c82a2))
* **p2p:** Updated libp2p dependencies
([#1792](https://github.com/AztecProtocol/aztec-packages/issues/1792))
([79df831](https://github.com/AztecProtocol/aztec-packages/commit/79df83134e15655dc3a5ed9dae00dc52a3d40681))
* Sandbox logging tweaks
([#1797](https://github.com/AztecProtocol/aztec-packages/issues/1797))
([0e3914e](https://github.com/AztecProtocol/aztec-packages/commit/0e3914ed6ad63062add1cc08f6ea85646c068f8a))
* split out yarn-project bootstrap.sh
([#1790](https://github.com/AztecProtocol/aztec-packages/issues/1790))
([1788fe6](https://github.com/AztecProtocol/aztec-packages/commit/1788fe6259f5e7fd191929b27996a7342e3f13e5))

---
This PR was generated with [Release
Please](https://github.com/googleapis/release-please). See
[documentation](https://github.com/googleapis/release-please#release-please).

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

docs: Wallet dev docs (#1746)

Developer docs on wallets. Introduces a new "architecture" section for
wallets, that elaborates on the RPC server and entrypoints. Deletes the
"building a wallet" tutorial, since building a wallet is excessively
complex for a tutorial.

Fixes #1741
Closes #1745
Pending #1744

git subrepo push --branch=main docs

subrepo:
  subdir:   "docs"
  merged:   "6f755743d"
upstream:
  origin:   "https://github.com/AztecProtocol/docs"
  branch:   "main"
  commit:   "6f755743d"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

feat: Update safe_math and move to libraries (#1803)

Fixes #1794 and address the wrong check in `mul`.

Would prefer to add tests directly, but noir don't support failing tests
in noir yet, so there is really no good reason to do that currently. See
https://github.com/noir-lang/noir/issues/1994

test: add browser test to canary flow (#1808)

Adding Aztec.js browser test to our canary flow to ensure published npm
package is stable

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

docs: Account contract tutorial (#1772)

Tutorial for writing an account contract. Includes tweaks to payload
helpers in aztec.js to make the process easier.

Fixes #1744
See also #1746

---------

Co-authored-by: Michael Connor <mike@aztecprotocol.com>

git subrepo push --branch=main docs

subrepo:
  subdir:   "docs"
  merged:   "2fd486a6c"
upstream:
  origin:   "https://github.com/AztecProtocol/docs"
  branch:   "main"
  commit:   "2fd486a6c"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

chore: fixed linter errors for `ecc`, `numeric` and `common` modules (#1714)

The majority of the barretenberg codebase does not conform to our C++
style guide rules.

This PR updates the `common`, `numeric` and `ecc` modules to conform to
the guide. These 3 modules should now produce no linter errors.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [x] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [x] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [x] Every change is related to the PR description.
- [x] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

---------

Co-authored-by: kevaundray <kevtheappdev@gmail.com>

git subrepo push --branch=master circuits/cpp/barretenberg

subrepo:
  subdir:   "circuits/cpp/barretenberg"
  merged:   "cca5c1bf1"
upstream:
  origin:   "https://github.com/AztecProtocol/barretenberg"
  branch:   "master"
  commit:   "cca5c1bf1"
git-subrepo:
  version:  "0.4.6"
  origin:   "???"
  commit:   "???"

feat: More reliable getTxReceipt api. (#1793)

Closes #1402 #1548

Previous approach had too many issues - only sender and the recipients
(after they decrypt their notes) can call this api. The data in the
receipt was not consistent.
And the role of TxDao is confusing. Delete it now and remove some
complicated code around it.

Remove the checklist to signal you've completed it. Enable auto-merge if
the PR is ready to merge.
- [ ] If the pull request requires a cryptography review (e.g.
cryptographic algorithm implementations) I have added the 'crypto' tag.
- [ ] I have reviewed my diff in github, line by line and removed
unexpected formatting changes, testing logs, or commented-out code.
- [ ] Every change is related to the PR description.
- [ ] I have
[linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
this pull request to relevant issues (if any exist).

dan-aztec committed:
add privatetoken.json for now for ease of local dev

add empty scripts

dan-aztec commented:
i did a bad squash, gonna leave this here in case i need something but opened fresh PR at https://github.com/AztecProtocol/aztec-packages/pull/1825
